### ISSUE TITLE
Variable names and meaning 

### ISSUE THREAD

**Issue by MostafaGomaa93** [EcoExtreML/STEMMUS_SCOPE#211]
Hi all,

Does anyone know the meaning of the "h", "hh" variables? The initial values of “h” and “hh” are set at lines 227->235 in the STEMMUS_SCOPE.m file

Thanks a lot in advance
Mostafa

**Comment by MostafaGomaa93** [comment]
got the answer from @yijianzeng by email

Both are for soil matric potential, the ‘hh’ is the one for the current time step, and ‘h’ is for the previous time step

**Comment by MostafaGomaa93** [comment]
Hi @yijianzeng @Yunfei-Wang1993 @Crystal-szj @EntingTang 
 I have spent some time going through the code, but it is a bit difficult to find what I need. Could anyone please help answer the following questions:

1. The soil water conservation equation (equation A13 in the [STEMMUS_SCOPE paper](https://gmd.copernicus.org/articles/14/1379/2021/gmd-14-1379-2021.html)), where are the calcutions of this equation?
2. What are the symbols used in the code for the following variables qLh, qLT, qLa, qvh, qvT, qva?
3. What do those symbols mean: `RHS`, `C4`, and `C4_a`? 
4. Why `RHS` is an output from different functions (e.g.  [solveSoilMoistureBalance](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Bsoilmoisture/solveSoilMoistureBalance.m#L14), [solveDryAirEquations](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Bdryair/solveDryAirEquations.m), [solveEnergyBalanceEquations](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/solveEnergyBalanceEquations.m), [calculateBoundaryConditions](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateBoundaryConditions.m#L29), [solveTridiagonalMatrixEquations](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Bdryair/solveTridiagonalMatrixEquations.m)) which confuses me

Thanks so much in advance

**Comment by MostafaGomaa93** [comment]
@SarahAlidoost @BSchilperoort 
To highlight what I need (because I am stuck for a long time so far), my first two questions from my comment above are related to the STEMMUS_SCOPE & MODFLOW coupling. In brief, what MODFLOW needs from STEMMUS_SCOPE is the "recharge flux", which is 
Recharge = sum of (qLh, qLT, qLa, qvh, qvT, qva) at the soil layer where groundwater level exist
I guess (need someone to confirm, mainly @yijianzeng) those variables are in the `calculateEnergyParameters.m` file in the following lines:
- QL at line [63](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L63), which is most likely = qLh + qLT, confirm?
- Qv at lines [70-72](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L70C1-L72C15), which is most likely = qVh + qVT, confirm?
- Qa at lines [64-66](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L70C1-L72C15), which is most likely = qLa, still can't figure out where it then qva, confirm?

**Comment by MostafaGomaa93** [comment]
Okay, i had a meeting with @yijianzeng which helps to answer my questions. 
> 
> * QL at line [63](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L63), which is most likely = qLh + qLT, confirm?
>
Yes
> * Qv at lines [70-72](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L70C1-L72C15), which is most likely = qVh + qVT, confirm?
>
Yes
> * Qa at lines [64-66](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L70C1-L72C15), which is most likely = qLa, confirm? still can't figure out where is then qva?
>
Currently in this version of the code so far (1.5), the airflow is neglected (see line [62](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L62)) and then qa = 0 (line [64](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L64)). However, the calcutions of qa is included (line [66](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Benergy/calculateEnergyParameters.m#L66)), but so far not used. The main reason for neglecting qa is because its effect (based on tests from Yijian and Liyanu) was small and it needs extra computational and time to be solved.
So in conclusion, the recharge flux that will be passed to MODFLOW will be of 4 components (qLh, qLT, qvh, qvT), only without (qLa and qva).

**Comment by MostafaGomaa93** [comment]
Hi @yijianzeng 
Please could you explain to me the [CalculateSoilwaterFlux](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/36a33008c98447515684f4e5fe57ebf99aad7544/src/%2Bsoilmoisture/calculatesSoilWaterFluxes.m#L7) function and its outputs? what are `SAVE`, `QMT` and `QMB`?
